xl/player/player.py

```diff
@@ -504,8 +571,31 @@ def engine_autoadvance_get_next_track(self, gapless=False):
             if self._auto_advance_delay != 0 or not self._gapless_enabled:
                 return
 
+        track = self._track
+        stop_at = track.get_tag_raw('__stopoffset') or 0
+        track_len = track.get_tag_raw('__length') or 0
+
+        if stop_at > track_len:
+            logger.debug('entering delayed stop')
+            diff = stop_at - track_len
+            self._paused_start = time.time()
+            self._paused_until = diff + self._paused_start
+            self._delayed_stop_id = GLib.timeout_add_seconds(1, self._wait_for_end)
```

Comment:
Why is this a periodic callback? Why not just delay for the difference?

Reply:
Was just for printing a pretty countdown in the log